sklearn/model_selection/_split.py

```diff
@@ -476,28 +476,23 @@ class GroupKFold(_BaseKFold):
     --------
     >>> import numpy as np
     >>> from sklearn.model_selection import GroupKFold
-    >>> X = np.array([[1, 2], [3, 4], [5, 6], [7, 8]])
-    >>> y = np.array([1, 2, 3, 4])
-    >>> groups = np.array([0, 0, 2, 2])
+    >>> X = np.array([[1, 2], [3, 4], [5, 6], [7, 8], [9, 10], [11, 12]])
+    >>> y = np.array([1, 2, 3, 4, 5, 6])
+    >>> groups = np.array([0, 0, 2, 2, 3, 3])
     >>> group_kfold = GroupKFold(n_splits=2)
     >>> group_kfold.get_n_splits(X, y, groups)
     2
     >>> print(group_kfold)
     GroupKFold(n_splits=2)
     >>> for train_index, test_index in group_kfold.split(X, y, groups):
-    ...     print("TRAIN:", train_index, "TEST:", test_index)
-    ...     X_train, X_test = X[train_index], X[test_index]
-    ...     y_train, y_test = y[train_index], y[test_index]
-    ...     print(X_train, X_test, y_train, y_test)
+    ...     train_group, test_group = groups[train_index], groups[test_index]
+    ...     print(f"TRAIN:\tIndex: {train_index}, Group: {train_group}")
+    ...     print(f"TEST:\tIndex: {test_index}, Group: {test_group}")
```

Comment:
What about including information on the fold to make it a little more clear what is happening?

```
Fold 1:
    train_index=[2 3], train_group=[2 2]
    test_index=[0 1 4 5], test_group=[0 0 3 3]
Fold 2:
    train_index=[0 1 4 5],  train_group=[0 0 3 3]
    test_index=[2 3], test_group=[2 2]
```

Reply:
I thought about this but couldn't think of a way to do it neatly, I got:
```python
    >>> for i, (train_index, test_index) in enumerate(group_kfold.split(X, y, groups)):
    ...     train_group, test_group = groups[train_index], groups[test_index]
    ...     print(f"Fold {i}")
    ...     print(f"\tTRAIN:\tIndex: {train_index}, Group: {train_group}")
    ...     print(f\t"TEST:\tIndex: {test_index}, Group: {test_group}")
```
WDYT?

Reply:
I personally don't mind this, and I don't think you can make that first line shorter without changing the variable names or doing something non-Pythonic. Another way would be the following, which produces what I had above &ndash; it makes the `print()` statements easier to follow, but the first line is the same:

```python
    >>> for i, (train_index, test_index) in enumerate(group_kfold.split(X, y, groups)):
    ...     train_group, test_group = groups[train_index], groups[test_index]
    ...     print(f"Fold {i + 1}")
    ...     print(f"\t{train_index=}, {train_group=}")
    ...     print(f"\t{test_index=}, {test_group=}")
```

Reply:
Thanks, happy to amend but would like to get some other opinions (maybe @thomasjpfan?) just because I would like to update all other similar example sections for other classes in `_split.py` (e.g., [GroupKFold](https://scikit-learn.org/dev/modules/generated/sklearn.model_selection.GroupKFold.html#sklearn.model_selection.GroupKFold)) to the same format.

Reply:
That also LGTM. Should it be `group=` to match `index=`?

Reply:
Yes it should be `group=`.